cmd/private-org-peribolos-sync/main_test.go

```diff
@@ -0,0 +1,277 @@
+			Description:      pntrString("Test Repo: repo1"),
+		},
+		"repo2": {
+			HasProjects:      pntrBool(false),
+			AllowSquashMerge: pntrBool(false),
+			AllowMergeCommit: pntrBool(false),
+			AllowRebaseMerge: pntrBool(false),
+			Description:      pntrString("Test Repo: repo2"),
+		},
+		"repo3": {
+			HasProjects:      pntrBool(false),
+			AllowSquashMerge: pntrBool(false),
+			AllowMergeCommit: pntrBool(false),
+			AllowRebaseMerge: pntrBool(false),
+			Description:      pntrString("Test Repo: repo3"),
+		},
+		"repo4": {
+			HasProjects:      pntrBool(false),
+			AllowSquashMerge: pntrBool(false),
+			AllowMergeCommit: pntrBool(false),
+			AllowRebaseMerge: pntrBool(false),
+			Description:      pntrString("Test Repo: repo4"),
+		},
+	}
+
+	repos := generateRepositories(fakeGitHubClient{}, orgRepos, logrus.WithField("destination-org", "testOrg"))
+	if !reflect.DeepEqual(repos, expectedRepos) {
+		t.Fatal(cmp.Diff(repos, expectedRepos))
+	}
+}
+
+func TestValidateWhitelist(t *testing.T) {
+	testCases := []struct {
+		id          string
+		whitelist   []string
+		errExpected bool
+	}{
+		{
+			id:        "repos to include, no error expected",
+			whitelist: []string{"testshift/origin"},
+		},
+		{
+			id:          "no valid repo format, error expected",
+			whitelist:   []string{"no-valid-repo"},
+			errExpected: true,
+		},
+		{
+			id:        "multiple repos to include, no error expected",
+			whitelist: []string{"openshift/repo1", "openshift/repo2", "testshift/repo1", "testshift/repo2"},
+		},
+		{
+			id:          "multiple repos to include, with mixed valid and non valid repos , error expected",
+			whitelist:   []string{"openshift/repo1", "no-valid-repo", "testshift/repo1", "no-valid-repo"},
+			errExpected: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.id, func(t *testing.T) {
+			err := validateWhitelist(tc.whitelist)
+			if err != nil && !tc.errExpected {
+				t.Fatalf("No error expected: %v", err)
+			}
+
+			if err == nil && tc.errExpected {
+				t.Fatal("Error expected, got nil")
+			}
+		})
+	}
+}
+
+func TestGetWhitelistByOrg(t *testing.T) {
+	testCases := []struct {
+		id        string
+		whitelist []string
+		expected  map[string]sets.String
+	}{
+		{
+			id:        "repos to include, changes expected",
+			whitelist: []string{"testshift/origin"},
+			expected:  map[string]sets.String{"testshift": sets.NewString("origin")},
+		},
+		{
+			id:        "multiple repos to include, changes expected",
+			whitelist: []string{"openshift/repo1", "openshift/repo2", "testshift/repo1", "testshift/repo2"},
+			expected: map[string]sets.String{
+				"openshift": sets.NewString("repo1", "repo2"),
+				"testshift": sets.NewString("repo1", "repo2"),
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.id, func(t *testing.T) {
+			whitelistByOrg := getWhitelistByOrg(tc.whitelist)
+
+			if !reflect.DeepEqual(whitelistByOrg, tc.expected) {
+				t.Fatal(cmp.Diff(whitelistByOrg, tc.expected))
```

Comment:
can we add in the test name and maybe a human message? failure and a diff is really not fun to read

Reply:
(think about someone breaking this test who has never touched this code before)

Reply:
sure.